### ISSUE TITLE
fluctuation in per batch time in ImageNet examples.

### ISSUE THREAD

**Issue by zym1010** [pytorch/examples#146]
I tried to train AlexNet from scratch, and compare the training time using PyTorch and using Caffe. I'm on a Pascal Titan X, using PyTorch 0.1.11 from pip.

It looks to me that the time taken to compute each batch varies, from 0.1s to 2s (worst case). I used 256 batch, and 22 workers. The data is fetched from a PCIe NVMe SSD, so IO should not be an issue (I think?).

Is this expected, or is this something that can be addressed? Thanks.

~~~
Epoch: [0][905/5005]    Time 1.317 (0.256)      Data 1.264 (0.168)      Loss 6.8791 (6.9048)    Prec@1 0.000 (0.097)    Prec@5 0.391 (0.498)
Epoch: [0][906/5005]    Time 0.099 (0.255)      Data 0.001 (0.168)      Loss 6.9001 (6.9048)    Prec@1 0.000 (0.096)    Prec@5 0.000 (0.498)
Epoch: [0][907/5005]    Time 0.103 (0.255)      Data 0.000 (0.168)      Loss 6.8702 (6.9047)    Prec@1 0.781 (0.097)    Prec@5 1.562 (0.499)
Epoch: [0][908/5005]    Time 0.102 (0.255)      Data 0.001 (0.167)      Loss 6.8882 (6.9047)    Prec@1 0.000 (0.097)    Prec@5 0.000 (0.498)
Epoch: [0][909/5005]    Time 0.257 (0.255)      Data 0.206 (0.167)      Loss 6.8977 (6.9047)    Prec@1 0.391 (0.097)    Prec@5 0.391 (0.498)
Epoch: [0][910/5005]    Time 0.102 (0.255)      Data 0.001 (0.167)      Loss 6.8973 (6.9047)    Prec@1 0.000 (0.097)    Prec@5 0.000 (0.498)
Epoch: [0][911/5005]    Time 0.603 (0.255)      Data 0.552 (0.168)      Loss 6.8929 (6.9047)    Prec@1 0.000 (0.097)    Prec@5 0.391 (0.498)
Epoch: [0][912/5005]    Time 0.101 (0.255)      Data 0.001 (0.168)      Loss 6.8911 (6.9047)    Prec@1 0.000 (0.097)    Prec@5 0.781 (0.498)
Epoch: [0][913/5005]    Time 0.497 (0.255)      Data 0.445 (0.168)      Loss 6.8757 (6.9046)    Prec@1 0.000 (0.097)    Prec@5 1.172 (0.499)
Epoch: [0][914/5005]    Time 0.111 (0.255)      Data 0.001 (0.168)      Loss 6.8713 (6.9046)    Prec@1 0.000 (0.097)    Prec@5 1.172 (0.499)
Epoch: [0][915/5005]    Time 0.100 (0.255)      Data 0.001 (0.167)      Loss 6.8851 (6.9046)    Prec@1 0.000 (0.097)    Prec@5 0.000 (0.499)
Epoch: [0][916/5005]    Time 0.106 (0.255)      Data 0.001 (0.167)      Loss 6.8716 (6.9045)    Prec@1 0.000 (0.097)    Prec@5 1.172 (0.500)
Epoch: [0][917/5005]    Time 0.156 (0.255)      Data 0.105 (0.167)      Loss 6.9136 (6.9046)    Prec@1 0.000 (0.097)    Prec@5 0.391 (0.500)
Epoch: [0][918/5005]    Time 0.102 (0.255)      Data 0.001 (0.167)      Loss 6.8948 (6.9045)    Prec@1 0.000 (0.096)    Prec@5 0.000 (0.499)
Epoch: [0][919/5005]    Time 0.101 (0.255)      Data 0.001 (0.167)      Loss 6.8860 (6.9045)    Prec@1 0.000 (0.096)    Prec@5 1.172 (0.500)
Epoch: [0][920/5005]    Time 0.101 (0.254)      Data 0.001 (0.167)      Loss 6.8774 (6.9045)    Prec@1 0.000 (0.096)    Prec@5 0.391 (0.500)
Epoch: [0][921/5005]    Time 0.108 (0.254)      Data 0.001 (0.167)      Loss 6.8833 (6.9045)    Prec@1 0.391 (0.097)    Prec@5 0.391 (0.500)
Epoch: [0][922/5005]    Time 0.163 (0.254)      Data 0.106 (0.166)      Loss 6.8969 (6.9045)    Prec@1 0.000 (0.096)    Prec@5 0.000 (0.499)
Epoch: [0][923/5005]    Time 0.251 (0.254)      Data 0.194 (0.166)      Loss 6.8844 (6.9044)    Prec@1 0.000 (0.096)    Prec@5 0.391 (0.499)
~~~

I noticed this because when I print timing info every 10 epochs, it seems that the average value is far from the value of the current batch.

**Comment by apaszke** [comment]
It's not expected, but I don't think it's a library bug. It might be a system configuration/hardware issue. Did you try running with fewer/more workers? AlexNet is so fast that it might be hard to keep up with data loading just because of limited disk bandwidth.

**Comment by zym1010** [comment]
@apaszke thanks. I think maybe AlexNet is the culprit, as the timing becomes more stable when using VGG.
